### PR TITLE
Minor: Move some aggregate error tests to sqllogictests

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

part of #4495 

# Rationale for this change

We are slowly moving tests to the easier to use / faster to run sqllogictest framework

# What changes are included in this PR?

port some tests 
# Are these changes tested?

Yes

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 1]
```diff
@@ -36,6 +36,53 @@ STORED AS CSV
...
+SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
+
+statement error  Error during planning: The weight argument for ApproxPercentileContWithWeight does not support inputs of type Utf8
+SELECT approx_percentile_cont_with_weight(c3, c1, 0.95) FROM aggregate_test_100
+
+statement error Error during planning: The percentile argument for ApproxPercentileContWithWeight must be Float64, not Utf8
+SELECT approx_percentile_cont_with_weight(c3, c2, c1) FROM aggregate_test_100
+
+
+# csv_query_approx_percentile_cont_with_weight
```
Comment (korowa) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:58]: Looks like duplicated block, except double spaces on [line 51](https://github.com/apache/arrow-datafusion/pull/5055/files#diff-d44e3605231ebbdef511cc254b4fe6d6d84af632275e98daefc155e3db0181efR51) and dot in the end of [line 48](https://github.com/apache/arrow-datafusion/pull/5055/files#diff-d44e3605231ebbdef511cc254b4fe6d6d84af632275e98daefc155e3db0181efR48)

Comment (korowa) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:58]: And it also seems each of these 3 error messages should end with dot, though sqllogictests completely  ignores actual error message - even
```
statement error just some random text here
SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
```
counts as passed test 🤔

Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:58]: Good call with duplicated block. Thank you 🙏 



As for the error messages, I am not sure what you mean by "though sqllogictests completely ignores actual error message - even"

I tried changing the expected error message like this

```diff
diff --git a/datafusion/core/tests/sqllogictests/test_files/aggregate.slt b/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
index 3b527dcd8..6931018da 100644
--- a/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
+++ b/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
@@ -56,7 +56,7 @@ SELECT approx_percentile_cont_with_weight(c3, c2, c1) FROM aggregate_test_100
 
 
 # csv_query_approx_percentile_cont_with_weight
-statement error Error during planning: The function ApproxPercentileContWithWeight does not support inputs of type Utf8.
+statement error some random text
 SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
 
 statement error Error during planning: The weight argument for ApproxPercentileContWithWeight does not support inputs of type Utf8
```

And then test failed, as I expected:
```shell
cargo test --test sqllogictests
....
STORED AS CSV
WITH HEADER ROW
LOCATION '../../testing/data/csv/aggregate_test_100.csv'"
[aggregate.slt] Running query: "SELECT approx_distinct(c9) count_c9, approx_distinct(cast(c9 as varchar)) count_c9_str FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c3, c1, 0.95) FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c3, c2, c1) FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100"
Error: statement is expected to fail with error:
	some random text
but got error:
	DataFusion error: Error during planning: The function ApproxPercentileContWithWeight does not support inputs of type Utf8.
[SQL] SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
at tests/sqllogictests/test_files/aggregate.slt:59

error: test failed, to rerun pass `-p datafusion --test sqllogictests`
```

🤔

**Review by melgenek — APPROVED** [review]

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt (hunk 1 above)
Comment (korowa) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:47]: Sorry, I should have checked this more careful - it was unsaved version of file.

Now I see that it's ok if expected result is leftmost substring of SQL execution result, so it perfectly explains why missing dot in the end of string doesn't affect test results.

**Review by korowa — APPROVED** [review]

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:47]: Thank you for double checking @korowa

**Review by jackwener — APPROVED** [review]
LGTM, thanks @alamb

**Pull request merged**